lib/rspec/core/example.rb

```diff
@@ -227,14 +227,14 @@ def run(example_group_instance, reporter)
               rescue Pending::SkipDeclaredInExample
                 # no-op, required metadata has already been set by the `skip`
                 # method.
-              rescue Exception => e
+              rescue Support::AllExceptionsExceptOnesWeMustNotRescue => e
```

Comment:
I don't understand why, but somehow this change causes a failure for [this cucumber scenario](https://github.com/rspec/rspec-core/blob/229c09cf59c1014c10afc67b190a8ae5ced111fa/features/configuration/failure_exit_code.feature#L40-L53) on 1.8.7:

https://travis-ci.org/rspec/rspec-core/jobs/77979590

If I change this back to `Exception` that passes.  Makes absolutely no sense to me :(.

Anyone got any ideas why?